dtfs-central-api/src/helpers/amendments/tfm-facility-latest-completed-amendment-to-facility-value.test.ts

```diff
@@ -0,0 +1,101 @@
+      };
+
+      // Act
+      const result = getLatestCompletedAmendmentToFacilityValue(tfmFacility);
+
+      // Assert
+      expect(result).toBeUndefined();
+    });
+
+    it('returns undefined when the supplied tfm facility amendments array is empty', () => {
+      // Arrange
+      const tfmFacility: TfmFacility = {
+        ...aTfmFacility(),
+        amendments: [],
+      };
+
+      // Act
+      const result = getLatestCompletedAmendmentToFacilityValue(tfmFacility);
+
+      // Assert
+      expect(result).toBeUndefined();
+    });
+
+    it('returns undefined when the supplied tfm facility has no completed amendments', () => {
+      // Arrange
+      const amendments: TfmFacilityAmendment[] = [
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.IN_PROGRESS, value: 100000 },
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.NOT_STARTED, value: 200000 },
+      ];
+
+      const tfmFacility: TfmFacility = {
+        ...aTfmFacility(),
+        amendments,
+      };
+
+      // Act
+      const result = getLatestCompletedAmendmentToFacilityValue(tfmFacility);
+
+      // Assert
+      expect(result).toBeUndefined();
+    });
+
+    it('returns undefined when the supplied tfm facility has completed amendments without a defined value', () => {
+      // Arrange
+      const amendments: TfmFacilityAmendment[] = [
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.COMPLETED, value: undefined },
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.COMPLETED, value: undefined },
+      ];
+
+      const tfmFacility: TfmFacility = {
+        ...aTfmFacility(),
+        amendments,
+      };
+
+      // Act
+      const result = getLatestCompletedAmendmentToFacilityValue(tfmFacility);
+
+      // Assert
+      expect(result).toBeUndefined();
+    });
+
+    it('returns the value of the latest completed amendment with a defined value', () => {
+      // Arrange
+      const latestCompletedAmendmentToFacilityValue = 5000;
+
+      const amendments: TfmFacilityAmendment[] = [
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.COMPLETED, value: 1, updatedAt: new Date('2022').getTime() },
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.NOT_STARTED, value: 2, updatedAt: new Date('2025').getTime() },
+        {
+          ...aTfmFacilityAmendment(),
+          status: AMENDMENT_STATUS.COMPLETED,
+          value: latestCompletedAmendmentToFacilityValue,
+          updatedAt: new Date('2023').getTime(),
+        },
+        { ...aTfmFacilityAmendment(), status: AMENDMENT_STATUS.IN_PROGRESS, value: 3, updatedAt: new Date('2024').getTime() },
+      ];
+
+      const tfmFacility: TfmFacility = {
+        ...aTfmFacility(),
+        amendments,
+      };
+
+      // Act
+      const result = getLatestCompletedAmendmentToFacilityValue(tfmFacility);
+
+      // Assert
+      expect(result).toEqual(latestCompletedAmendmentToFacilityValue);
+    });
```

Comment:
could we add a test with multiple completed amendments please?

Reply:
This test has multiple completed amendments